lightning/src/chain/chainmonitor.rs

```diff
@@ -262,6 +346,61 @@ where C::Target: chain::Filter,
 		self.monitors.write().unwrap().remove(funding_txo).unwrap().monitor
 	}
 
+	/// Indicates the persistence of a [`ChannelMonitor`] has completed after
+	/// [`ChannelMonitorUpdateErr::TemporaryFailure`] was returned from an update operation.
+	///
+	/// Thus, the anticipated use is, at a high level:
+	///  1) This [`ChainMonitor`] calls [`Persist::update_persisted_channel`] which stores the
+	///     update to disk and begins updating any remote (e.g. watchtower/backup) copies,
+	///     returning [`ChannelMonitorUpdateErr::TemporaryFailure`],
+	///  2) once all remote copies are updated, you call this function with the
+	///     `completed_update_id` that completed, and once all pending updates have completed the
+	///     channel will be re-enabled.
+	///
+	/// Returns an [`APIError::APIMisuseError`] if `funding_txo` does not match any currently
+	/// registered [`ChannelMonitor`]s.
+	pub fn channel_monitor_updated(&self, funding_txo: OutPoint, completed_update_id: MonitorUpdateId) -> Result<(), APIError> {
+		let monitors = self.monitors.read().unwrap();
+		let monitor_data = if let Some(mon) = monitors.get(&funding_txo) { mon } else {
+			return Err(APIError::APIMisuseError { err: format!("No ChannelMonitor matching funding outpoint {:?} found", funding_txo) });
+		};
+		let mut pending_monitor_updates = monitor_data.pending_monitor_updates.lock().unwrap();
+		pending_monitor_updates.retain(|update_id| *update_id != completed_update_id);
```

Comment:
> Should we be able to discard any updates that are <= the completed_update_id? Thought that was how the API worked prior to this PR

The change was a conscious decision (and I think is in the docs, but lmk if it can be more clear) - as the thinking on this has crystalized more towards "async update" not "temp failure" it makes sense to track all in-flight updates imo. If you map this to an actual async operation (eg via tokio) then there's some stream of updates, each which is put in a future, updated maybe with a remote server, and then we call updated here. If a newer update runs ahead and finishes before an older one, we still want to wait for the older one to complete before restoring the channel to normal operation.

> If not, should we debug_assert or so that at a monitor_update was removed? Is there any risk/side effects if a user calls this multiple times with the same completed_update_id?

Hmm, I mean we don't really need to "support" it, but also don't think we need to consider it a bug or a user failure/API violation?